R/req-chunk.R

```diff
@@ -0,0 +1,100 @@
+#' Chunk a request
+#'
+#' Use `req_chunk()` to specify how to request a chunk of data.
+#' Use `req_perform_multi()` to request all chunks.
+#' If you need more control use a combination of [req_perform()] and
+#' [multi_next_request()] to iterate through the chunks yourself.
+#'
+#' @inheritParams req_perform
+#' @param chunk_size The size of each chunk.
+#' @param data The vector to chunk.
+#' @param apply_chunk A function that applies the chunk to the request. It
+#'   takes two arguments:
+#'
+#'   1. `req`: the original request.
+#'   2. `chunk`: the current data chunk.
+#' @param parse_resp A function with one argument `resp` that parses the
+#'   response.
+#' @returns A modified HTTP [request].
+#' @export
+#'
+#' @examples
+#' base_req <- request("https://api.restful-api.dev/objects")
+#'
+#' ids <- 1:7
+#' chunk_size <- 3
+#'
+#' apply_chunk <- function(req, chunk) {
+#'   chunk <- rlang::set_names(chunk, "id")
+#'   req_url_query(req, !!!chunk)
+#' }
+#'
+#' parse_resp <- function(resp) {
+#'   parsed <- resp_body_json(resp)
+#'   data.frame(
+#'     id = sapply(parsed, function(x) x$id),
+#'     name = sapply(parsed, function(x) x$name)
+#'   )
+#' }
+#'
+#' \dontrun{
+#' req <- req_chunk(
+#'   base_req,
+#'   chunk_size = 3,
+#'   data = ids,
+#'   apply_chunk = apply_chunk,
+#'   parse_resp = parse_resp
+#' )
+#'
+#' responses <- req_perform_multi(req)
+#' }
+req_chunk <- function(req,
```

Comment:
It might be nice to have a helper to create a list of all requests so that one can perform them in parallel.
This would also work for two pagination patterns (pagination via offset and via page index). But this should be handled in a separate PR.